File: Chapter4/Python/Example4.py
# Chapter 4 - Poisson Distribution
# Example 4: Simulation  Poisson(6) — 1000 observations

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import poisson

# Parameter
lam             = 6
num_simulations = 1000

# Simulation
np.random.seed(42)
simulated_data = poisson.rvs(lam, size=num_simulations)

# Theoretical distribution
x               = np.arange(0, 21)
theoretical_pmf = poisson.pmf(x, lam)

# Plot
plt.figure(figsize=(12, 6))

plt.hist(simulated_data, bins=np.arange(-0.5, 20.5, 1),
         density=True, alpha=0.6, color='green',
         edgecolor='black', label='Simulation')

plt.plot(x, theoretical_pmf, 'ro-', linewidth=2,
         markersize=6, label='Theoretical Distribution')

plt.xlabel('Number of Events', fontsize=12)
plt.ylabel('Frequency / Probability', fontsize=12)
plt.title(f'Simulation vs Theoretical: Poisson({lam})', fontsize=14)
plt.legend(fontsize=11)
plt.grid(alpha=0.3)
plt.tight_layout()
plt.show()

# Statistics comparison
print(f"Simulation Mean:       {np.mean(simulated_data):.2f}")
print(f"Theoretical Mean:      {lam:.2f}")
print(f"Simulation Std Dev:    {np.std(simulated_data, ddof=1):.2f}")
print(f"Theoretical Std Dev:   {np.sqrt(lam):.2f}")

File: Chapter4/R/Example4.R
# Chapter 4 - Poisson Distribution
# Example 4: Simulation  Poisson(6) — 1000 observations

# Parameter
lambda          <- 6
num_simulations <- 1000

# Simulation
set.seed(42)
simulated_data <- rpois(num_simulations, lambda)

# Theoretical distribution
x               <- 0:20
theoretical_pmf <- dpois(x, lambda)

# Plot
hist(simulated_data, breaks = seq(-0.5, 20.5, 1),
     freq   = FALSE, col = 'lightgreen', border = 'black',
     xlab   = 'Number of Events',
     ylab   = 'Frequency / Probability',
     main   = paste0('Simulation vs Theoretical: Poisson(', lambda, ')'))

# Overlay theoretical line
lines(x, theoretical_pmf, col = 'red', lwd = 2, type = 'b', pch = 19)

legend('topright',
       legend = c('Simulation', 'Theoretical Distribution'),
       fill   = c('lightgreen', NA), border = c('black', NA),
       col    = c(NA, 'red'), lwd = c(NA, 2), pch = c(NA, 19))

# Statistics comparison
cat(sprintf("Simulation Mean:       %.2f\n", mean(simulated_data)))
cat(sprintf("Theoretical Mean:      %.2f\n", lambda))
cat(sprintf("Simulation Std Dev:    %.2f\n", sd(simulated_data)))
cat(sprintf("Theoretical Std Dev:   %.2f\n", sqrt(lambda)))
